### ISSUE TITLE
gnutls_handshake failed for stretch/main amd64

### ISSUE THREAD

**Issue by 0xpr03** [nodesource/distributions#504]
I've imported via the script, still:

```
apt-get update
Hit:1 http://security.debian.org stretch/updates InRelease
Hit:2 http://mirror2.hs-esslingen.de/mariadb/repo/10.2/debian stretch InRelease                                                                                                                                    
Hit:3 http://ppa.launchpad.net/webupd8team/java/ubuntu trusty InRelease                                                                                                                                            
Hit:4 http://ftp.debian.org/debian stretch-backports InRelease                                 
Ign:5 http://ftp.de.debian.org/debian stretch InRelease                  
Hit:6 http://ftp.de.debian.org/debian stretch-updates InRelease
Hit:7 http://ftp.de.debian.org/debian stretch Release
Ign:8 https://deb.nodesource.com/node_7.x stretch InRelease
Ign:9 https://deb.nodesource.com/node_7.x stretch Release
Ign:10 https://deb.nodesource.com/node_7.x stretch/main all Packages
Ign:11 https://deb.nodesource.com/node_7.x stretch/main amd64 Packages
Ign:12 https://deb.nodesource.com/node_7.x stretch/main i386 Packages
Ign:13 https://deb.nodesource.com/node_7.x stretch/main Translation-en_US
Ign:14 https://deb.nodesource.com/node_7.x stretch/main Translation-en
Ign:10 https://deb.nodesource.com/node_7.x stretch/main all Packages
Ign:11 https://deb.nodesource.com/node_7.x stretch/main amd64 Packages
Ign:12 https://deb.nodesource.com/node_7.x stretch/main i386 Packages
Ign:13 https://deb.nodesource.com/node_7.x stretch/main Translation-en_US
Ign:14 https://deb.nodesource.com/node_7.x stretch/main Translation-en
Ign:10 https://deb.nodesource.com/node_7.x stretch/main all Packages
Ign:11 https://deb.nodesource.com/node_7.x stretch/main amd64 Packages
Ign:12 https://deb.nodesource.com/node_7.x stretch/main i386 Packages
Ign:13 https://deb.nodesource.com/node_7.x stretch/main Translation-en_US
Ign:14 https://deb.nodesource.com/node_7.x stretch/main Translation-en
Ign:10 https://deb.nodesource.com/node_7.x stretch/main all Packages
Ign:11 https://deb.nodesource.com/node_7.x stretch/main amd64 Packages
Ign:12 https://deb.nodesource.com/node_7.x stretch/main i386 Packages
Ign:13 https://deb.nodesource.com/node_7.x stretch/main Translation-en_US
Ign:14 https://deb.nodesource.com/node_7.x stretch/main Translation-en
Ign:10 https://deb.nodesource.com/node_7.x stretch/main all Packages
Ign:11 https://deb.nodesource.com/node_7.x stretch/main amd64 Packages
Ign:12 https://deb.nodesource.com/node_7.x stretch/main i386 Packages
Ign:13 https://deb.nodesource.com/node_7.x stretch/main Translation-en_US
Ign:14 https://deb.nodesource.com/node_7.x stretch/main Translation-en
Ign:10 https://deb.nodesource.com/node_7.x stretch/main all Packages
Err:11 https://deb.nodesource.com/node_7.x stretch/main amd64 Packages
  gnutls_handshake() failed: Public key signature verification has failed.
Ign:12 https://deb.nodesource.com/node_7.x stretch/main i386 Packages
Ign:13 https://deb.nodesource.com/node_7.x stretch/main Translation-en_US
Ign:14 https://deb.nodesource.com/node_7.x stretch/main Translation-en
Reading package lists... Done
W: The repository 'https://deb.nodesource.com/node_7.x stretch Release' does not have a Release file.
N: Data from such a repository can't be authenticated and is therefore potentially dangerous to use.
N: See apt-secure(8) manpage for repository creation and user configuration details.
E: Failed to fetch https://deb.nodesource.com/node_7.x/dists/stretch/main/binary-amd64/Packages  gnutls_handshake() failed: Public key signature verification has failed.
E: Some index files failed to download. They have been ignored, or old ones used instead.
```

**Comment by 0xpr03** [comment]
Tried with manual method & before via
```
curl -sL https://deb.nodesource.com/setup_7.x | bash -
apt-get install -y nodejs
```
same error.

**Comment by chrislea** [comment]
There is definitely a Release file for stretch, it's here:

https://deb.nodesource.com/node_7.x/dists/stretch/Release

Can you see it in a browser? Also, Node v7 is basically EOL already, you should update to Node v8 immediately.

**Comment by 0xpr03** [comment]
I can see it inside my browser, it doesn't matter whether I'm using node_8.x or 7.x.
Just for the record:
This was just based upon [gitlab source install](https://docs.gitlab.com/ce/install/installation.html#4-node) due to my little problem with the mysql connector after a dist-upgrade to stretch.. (so essentially I'm trying to fix my gitlab installation ;) )

**Comment by 0xpr03** [comment]
But essentially its gnutls_handshake complaining about pks failing, so I'm not sure whether this has anythign to do with the release file or not.

**Comment by 0xpr03** [comment]
```
apt-get update
Hit:1 http://security.debian.org stretch/updates InRelease
Hit:2 http://ppa.launchpad.net/webupd8team/java/ubuntu trusty InRelease
Hit:3 http://mirror2.hs-esslingen.de/mariadb/repo/10.2/debian stretch InRelease              
Hit:4 http://ftp.debian.org/debian stretch-backports InRelease                               
Ign:5 http://ftp.de.debian.org/debian stretch InRelease                                      
Hit:6 http://ftp.de.debian.org/debian stretch-updates InRelease
Hit:7 http://ftp.de.debian.org/debian stretch Release
Ign:8 https://deb.nodesource.com/node_9.x stretch InRelease
Ign:9 https://deb.nodesource.com/node_9.x stretch Release
Ign:10 https://deb.nodesource.com/node_9.x stretch/main amd64 Packages
Ign:11 https://deb.nodesource.com/node_9.x stretch/main i386 Packages
Ign:12 https://deb.nodesource.com/node_9.x stretch/main all Packages
Ign:13 https://deb.nodesource.com/node_9.x stretch/main Translation-en_US
Ign:14 https://deb.nodesource.com/node_9.x stretch/main Translation-en
Ign:10 https://deb.nodesource.com/node_9.x stretch/main amd64 Packages
Ign:11 https://deb.nodesource.com/node_9.x stretch/main i386 Packages
Ign:12 https://deb.nodesource.com/node_9.x stretch/main all Packages
Ign:13 https://deb.nodesource.com/node_9.x stretch/main Translation-en_US
Ign:14 https://deb.nodesource.com/node_9.x stretch/main Translation-en
Ign:10 https://deb.nodesource.com/node_9.x stretch/main amd64 Packages
Ign:11 https://deb.nodesource.com/node_9.x stretch/main i386 Packages
Ign:12 https://deb.nodesource.com/node_9.x stretch/main all Packages
Ign:13 https://deb.nodesource.com/node_9.x stretch/main Translation-en_US
Ign:14 https://deb.nodesource.com/node_9.x stretch/main Translation-en
Ign:10 https://deb.nodesource.com/node_9.x stretch/main amd64 Packages
Ign:11 https://deb.nodesource.com/node_9.x stretch/main i386 Packages
Ign:12 https://deb.nodesource.com/node_9.x stretch/main all Packages
Ign:13 https://deb.nodesource.com/node_9.x stretch/main Translation-en_US
Ign:14 https://deb.nodesource.com/node_9.x stretch/main Translation-en
Ign:10 https://deb.nodesource.com/node_9.x stretch/main amd64 Packages
Ign:11 https://deb.nodesource.com/node_9.x stretch/main i386 Packages
Ign:12 https://deb.nodesource.com/node_9.x stretch/main all Packages
Ign:13 https://deb.nodesource.com/node_9.x stretch/main Translation-en_US
Ign:14 https://deb.nodesource.com/node_9.x stretch/main Translation-en
Err:10 https://deb.nodesource.com/node_9.x stretch/main amd64 Packages
  gnutls_handshake() failed: Public key signature verification has failed.
Ign:11 https://deb.nodesource.com/node_9.x stretch/main i386 Packages
Ign:12 https://deb.nodesource.com/node_9.x stretch/main all Packages
Ign:13 https://deb.nodesource.com/node_9.x stretch/main Translation-en_US
Ign:14 https://deb.nodesource.com/node_9.x stretch/main Translation-en
Reading package lists... Done
W: The repository 'https://deb.nodesource.com/node_9.x stretch Release' does not have a Release file.
N: Data from such a repository can't be authenticated and is therefore potentially dangerous to use.
N: See apt-secure(8) manpage for repository creation and user configuration details.
E: Failed to fetch https://deb.nodesource.com/node_9.x/dists/stretch/main/binary-amd64/Packages  gnutls_handshake() failed: Public key signature verification has failed.
E: Some index files failed to download. They have been ignored, or old ones used instead
```

**Comment by 0xpr03** [comment]
Other information: complaining about https://deb.nodesource.com/node_8.x/dists/stretch/main/source/Sources which does send a 404

**Comment by 0xpr03** [comment]
Update: It works on my virtual machine, but it won't on my server. Same configuration, updates etc.

**Comment by 0xpr03** [comment]
Fixed, issue with package dependencies of dmm.

**Comment by saschaludwig** [comment]
I have the same problem here with nodejs 10:

    Err:6 https://deb.nodesource.com/node_10.x stretch Release
      gnutls_handshake() failed: Public key signature verification has failed.

**Comment by Butterfly** [comment]
> I have the same problem here with nodejs 10 on debian stretch:
> 
> ```
> Err:6 https://deb.nodesource.com/node_10.x stretch Release
>   gnutls_handshake() failed: Public key signature verification has failed.
> ```

Mine's similarly different these few months later on the upcoming next Stable, Buster. Verbiage is different, but the message is still that the public key verification is the issue:

```
W: GPG error: https://deb.nodesource.com/node_10.x buster InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 1655A0AB68576280
E: The repository 'https://deb.nodesource.com/node_10.x buster InRelease' is not signed.
```

I know that we can add the key. I've done it. Every 3 years or so, this comes up (so I forget HOW to).

I'm using `yarn` to *attempt* to build `beakerbrowser`. `yarn` instruction helps out with this:

`curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | sudo apt-key add -`

The reason I'm on this thread this second is because I was out searching the Net hoping that a trustworthy source shared that same for `nodesource`. Am now wandering off ALSO hoping that posting `apt`/`apt-get`'s feedback about apt-key potential helps someone else... some day. :smile:

**Comment by Butterfly** [comment]
> I'm using `yarn` to _attempt_ to build `beakerbrowser`. `yarn` instruction helps out with this:
> 
> `curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | sudo apt-key add -`
> 
A search engine very cordially twisted my "nodesource" + "apt-key" keywords just enough to land a gpg key URL via [StackExchange](https://unix.stackexchange.com/questions/356735/problem-with-gpg-when-using-apt).

**IT WORKS!**

I tweaked `yarn`'s command to read:

`curl -sS https://deb.nodesource.com/gpgkey/nodesource.gpg.key |sudo apt-key add -`

Then reran NodeSource's recommended:

`curl -sL https://deb.nodesource.com/setup_10.x | bash -`

It worked as expected **EXCEPT that it THEN** said... It was adding the **SAME key** using the **SAME command** that was initially issued to get it to work in the first place. Its small tweak was to use `curl -s` instead of `curl -sS`.

So somehow for some part of the public key verification issue some of us have, it appears to be the order in which the commands are [enacted].

Am off to install now. *yay!*

PS Maybe some day the... captcha will work long enough that I could sign up for the newsletter.....? :smile:
